pkg/api/folder.go

```diff
@@ -51,7 +52,13 @@ func (hs *HTTPServer) GetFolderByUID(c *models.ReqContext) response.Response {
 
 func (hs *HTTPServer) GetFolderByID(c *models.ReqContext) response.Response {
 	s := dashboards.NewFolderService(c.OrgId, c.SignedInUser, hs.SQLStore)
-	folder, err := s.GetFolderByID(c.Req.Context(), c.ParamsInt64(":id"))
+
+	id, err := strconv.ParseInt(web.Params(c.Req)[":id"], 10, 64)
```

Comment:
Id of zero should be allowed here because that is the "general" folder.

http://localhost:3000/api/folders/id/0 gets me the general folder before these changes.